src/Airship.Net.MessageCenter/Platforms/iOS/Modules/AirshipMessageCenter.cs

```diff
@@ -174,7 +199,32 @@ public Task DisplayMessage(string messageId)
             return epoch.AddSeconds(date.SecondsSince1970);
         }
 
-        // Event handlers will be added in a future phase to maintain existing event patterns
-        // For now, users can still use the legacy events through Airship.Instance if needed
+        private EventHandler<MessageCenterEventArgs>? onMessageCenterDisplay;
+
+        /// <summary>
+        /// Add/remove the Message Center display listener.
+        /// </summary>
+        public event EventHandler<MessageCenterEventArgs> OnMessageCenterDisplay
+        {
+            add
+            {
+                onMessageCenterDisplay += value;
+                AirshipMessageCenterDisplayDelegate messageCenterDisplayDelegate = new AirshipMessageCenterDisplayDelegate((messageId) =>
+                     {
+                         onMessageCenterDisplay?.Invoke(this, new MessageCenterEventArgs(messageId));
+                     });
+                UAirship.MessageCenter.WeakDisplayDelegate = messageCenterDisplayDelegate;
```

Comment:
This is weak so it will get cleaned up right? We need to store a strong reference to it